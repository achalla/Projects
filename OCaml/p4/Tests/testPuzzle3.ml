(* Test Puzzle Functions 3  *)

#use "puzzle.ml";;

let a = [0;1;2;3];;
let b = [1;0;2;3];;

print_endline "single_move [[a]]" ;;
List.map prt_int_list_list (List.sort compare (single_move [[a]])) ;;

print_endline "single_move [[a;b]]" ;;
List.map prt_int_list_list (List.sort compare (single_move [[a;b]])) ;;

print_endline "single_move [[b;a]]" ;;
List.map prt_int_list_list (List.sort compare (single_move [[b;a]])) ;;

print_endline "single_move [[a];[b]]" ;;
List.map prt_int_list_list (List.sort compare (single_move [[a];[b]])) ;;

print_endline "single_move [[b];[a]]" ;;
List.map prt_int_list_list (List.sort compare (single_move [[b];[a]])) ;;
