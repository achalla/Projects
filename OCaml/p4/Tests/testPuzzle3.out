single_move [[a]]
[1;0;2;3][0;1;2;3]
[2;1;0;3][0;1;2;3]
single_move [[a;b]]
[1;0;2;3][0;1;2;3][1;0;2;3]
[2;1;0;3][0;1;2;3][1;0;2;3]
single_move [[b;a]]
[0;1;2;3][1;0;2;3][0;1;2;3]
[1;3;2;0][1;0;2;3][0;1;2;3]
single_move [[a];[b]]
[0;1;2;3][1;0;2;3]
[1;0;2;3][0;1;2;3]
[1;3;2;0][1;0;2;3]
[2;1;0;3][0;1;2;3]
single_move [[b];[a]]
[0;1;2;3][1;0;2;3]
[1;0;2;3][0;1;2;3]
[1;3;2;0][1;0;2;3]
[2;1;0;3][0;1;2;3]
